Source/quests.h

```diff
@@ -87,7 +87,7 @@ void InitQuests();
 void InitialiseQuestPools(uint32_t seed, QuestStruct quests[]);
 void CheckQuests();
 bool ForceQuests();
-bool QuestStatus(int i);
+bool QuestStatus(QuestStruct &quest, QuestDataStruct &questData);

```

Comment:
I don't think this change really improves reliability/readability since the caller now has to make sure to get the two corresponding structs and correctly use the same enum/offset twice.

Instead what might make sens would be to have QuestStatus() take the enum instead of an int:
```suggestion
bool QuestStatus(quest_id &questId);
```

Reply:
I had to change from `quest_id &questId` to `quest_id questId` because I kept getting below error message and was not sure how to resolve other than changing.

```
error: cannot bind non-const lvalue reference of type ‘devilution::quest_id&’ to an rvalue of type ‘devilution::quest_id’
 4966 |  if (QuestStatus(Q_VEIL) && monster.mtalkmsg >= TEXT_VEIL9) {
```

Reply:
sorry that was a typo on my side, simple value shouldn't be passed by reference :)